lib/model/rofolder.go

```diff
@@ -8,151 +8,88 @@ package model
 
 import (
 	"fmt"
-	"math/rand"
 	"time"
 
+	"github.com/syncthing/syncthing/lib/config"
 	"github.com/syncthing/syncthing/lib/sync"
 )
 
 type roFolder struct {
-	stateTracker
-
-	folder    string
-	intv      time.Duration
-	timer     *time.Timer
-	model     *Model
-	stop      chan struct{}
-	scanNow   chan rescanRequest
-	delayScan chan time.Duration
+	folder
 }
 
-type rescanRequest struct {
-	subs []string
-	err  chan error
-}
-
-func newROFolder(model *Model, folder string, interval time.Duration) *roFolder {
+func newROFolder(model *Model, cfg config.FolderConfiguration) *roFolder {
 	return &roFolder{
-		stateTracker: stateTracker{
-			folder: folder,
-			mut:    sync.NewMutex(),
+		folder: folder{
+			stateTracker: stateTracker{
+				folderID: cfg.ID,
+				mut:      sync.NewMutex(),
+			},
+			scan: folderscan{
```

Comment:
This kind of "deep" construction isn't super clean in my eyes. The nice thing about refactoring out the `folderscan` etc is that it buys us some encapsulation. I know we already do this for the stateTracker, so there is a precedent, but I would prefer to have a function `newFolderscan() *folderscan` construct the type instead of naming all the attributes here. Possibly the type could be named `folderScanner` or so as well?